client/src/theme/breakpoints.ts

```diff
@@ -1,7 +1,13 @@
+// Note the following breakpoints for different devices:
+// - mobile: 479px and below
+// - tablet: 480-1439px
+// - desktop: 1440px and above
 export const breakpoints = {
   xs: '360px',
   sm: '480px',
   md: '768px',
   lg: '1024px',
   xl: '1440px',
+  tablet: '480px', // should be same as sm
+  desktop: '1440px', // should be same as xl
 }
```

Comment:
if the devices are still coupled to the sizes, perhaps the following would allow for easier maintenance and self-documentation:

```ts
const sizeBasedBreakpoints = {
  xs: '360px',
  sm: '480px',
  md: '768px',
  lg: '1024px',
  xl: '1440px',
}

export const breakpoints = {
  ...sizeBasedBreakpoints,
  tablet: sizeBasedBreakpoints.sm,
  desktop: sizeBasedBreakpoints.xl,
}

Reply:
Hm I think either way works? I did so because I recall using a check for device when using Vue in the past instead of pixel sizes and thought this would be easier for development as it reduces cognitive load. In the iPad / iPad Mini / iPad Pro use case where we need more granularity, I think we can just use the normal breakpoints then?

Reply:
For reference Vue's UI library, Vuetify, allows users to easily check if the device is mobile through its breakpoint service. For more information: https://vuetifyjs.com/en/features/breakpoints/#breakpoint-service-object

Reply:
I see, so Vue takes this approach. Do you see equivalent practice in React?

Reply:
Kind of? There are packages like [react-device-detect](https://github.com/duskload/react-device-detect) which help with this. There are 1.4k stars and 82 forks on the repository so its relatively popular as well.

Reply:
Tangential at best. @karrui - any opinions on breakpoint labels?

Reply:
I'd say stick with sizing; xs -> xxl sizes? This allows [`chakra#useBreakpointValue`](https://chakra-ui.com/docs/hooks/use-breakpoint-value) or the [responsive syntax](https://chakra-ui.com/docs/features/responsive-styles) to be more intuitive imho. Since breakpoints are sorted in ascending values, if there are two of the same sizes i.e. `sizes.sm` and `sizes.tablet` for example, this may break how the responsive syntax is parsed.

Reply:
Ah, I guess that's true if we use the array way. Okay then, let's just close this PR.

Reply:
To add on, I'll just some comments about the breakpoints for mobile/ tablet/ desktop as they are the breakpoints that we are considering in the design system.

Reply:
nonetheless, if you are only centered around the few breakpoints and if you really want to only call the breakpoints via their synthetic name, you can always declare the breakpoints (and override the defaults like this:)

```tsx
createBreakpoints({
  'base': '0px',
  'mobile': '?px',
  'tablet': '?px',
  'desktop': '?px',
  // Default breakpoints all "unused" with this hack
  'md': '9999px',
  'sm': '9999px',
  'lg': '9999px',
  'xl': '9999px',
  '2xl': '9999px',
});
```

this will allow the usage of both array and object responsive syntax, though now the developers has to make sure not to use any of the "default" breakpoints as reference (or their styles won't get applied until 9999px)

Reply:
Hm, I don't think it's a good idea to make the default breakpoints useless in the case where we need finer granularity in the future (as mentioned by @LoneRifle). Thanks for the 'hack' though!